### ISSUE TITLE
Cost APIs to indicate tags are available

### ISSUE THREAD

**Issue by dlabrecq** [project-koku/koku#1534]
**Describe the bug**
As a user, I want the UI's details page to disable the "View all tags" option when there are no tags to display. When there are no tags, this results in an empty modal dialog being shown unnecessarily.

In order to disable the "View all tags" option, the UI must pre-fetch tags for each row of the details table (e.g., representing each account). For a table with 10 rows, that would be 10 extra requests, which may not be necessary if the user never chooses the option.

Currently, the UI lazily requests tag data when a table row is expanded. Or, only when the "View all tags" option is selected.

Note that the "View all tags" option is disabled when the group_by is service or region. The UI only makes this option available for group_by accounts.

**To Reproduce**
1. In the Koku UI, go to 'Cloud details'
2. Click on 'Amazon Web Services'
3. Click on the kebab menu
4. See the "View all tags" option

**Expected behavior**
It would be ideal if the cost APIs provided an indication / count when tags are available. Then the UI wouldn't need to make extra requests to discover that.

**Screenshots**
This is an example of the additional API request made to fetch tags:
/v1/tags/aws/?filter[account]=Ladas-Test&filter[resolution]=monthly&filter[time_scope_units]=month&filter[time_scope_value]=-1

![Screen Shot 2019-12-10 at 10.52.05 PM.png](https://images.zenhubusercontent.com/5b47705f01db7d22de6439cd/f9ba70d5-df35-48f9-8233-b2a893025a5f)

This is a typical request made to fetch table data, which provides no indication that there are tags available.

/v1/reports/aws/costs/?delta=cost&filter[limit]=10&filter[offset]=0&filter[resolution]=monthly&filter[time_scope_units]=month&filter[time_scope_value]=-1&group_by[account]=*&order_by[cost]=desc

![Screen Shot 2019-12-10 at 10.56.10 PM.png](https://images.zenhubusercontent.com/5b47705f01db7d22de6439cd/564daf3d-a6f1-47d0-bf90-47d4c6e93f6c)

**Comment by chambridge** [comment]
I don't understand how this is a bug versus a feature request.

**Comment by adberglund** [comment]
@chargio This needs product input and prioritization before engineering picks this up. From our initial assessment this feels lower priority then what we currently have in the back log. 


-- Perhaps for the time being the UI could have a cleaner empty state when there are no tags?

**Comment by dlabrecq** [comment]
In the past, I was asked to mark new issues as a bug so they would be evaluated by the backlog triage -- it's a feature

**Comment by Red-HAP** [comment]
Initial solution is a stop-gap. Please do not close this issue as a second branch may be able to better solve the issue. Also, there may be some view-work that can be later attached to this ticket.

Refer questiond to @adberglund

**Comment by lcouzens** [comment]
Verified work so far, pushing back to backlog to iterate on other endpoints + make perf improvements.

**Comment by nlcwong** [comment]
@dlabrecq Yes, number of tags we would need to show, i'm trying to migrate the designs to the way the rest of the apps are displaying their tags (tag icon + number of tags which would link to a fuller list on a modal dialog).  Not necessary that we need a tag column in the tables, just the ability to show the tag count next to an icon somewhere in the expand section of a detail list.
